source/features/previous-version.tsx

```diff
@@ -0,0 +1,117 @@
+import * as pageDetect from 'github-url-detection';
+import React from 'dom-chef';
+import cache from 'webext-storage-cache';
+
+import features from '../feature-manager';
+import * as api from '../github-helpers/api';
+import observe from '../helpers/selector-observer';
+
+import GitHubURL from '../github-helpers/github-url';
+
+const getHistoryOids = cache.function('previous-version', async (githubUrl: GitHubURL): Promise<string[] | false> => {
+	const {resource: {history}} = await api.v4(`
+		resource(url: "/${githubUrl.user}/${githubUrl.repository}/commit/${githubUrl.branch}") {
+			... on Commit {
+				history(path: "${githubUrl.filePath}") {
+					nodes {
+						oid
+					}
+				}
+			}
+		}
+	`);
+
+	const nodes = history.nodes as any[];
+
+	if (nodes.length === 0) {
+		return false;
+	}
+
+	return nodes.map<string>(n => n.oid);
```

Comment:
This "string" here sets the type of `n`, which isn't right since it's an object

Reply:
Hmm.. The generic of Array.map means return type, not input type
But I just changed this to `nodes.map(node => node.oid as string);`